src/Chainweb/Pact/PactService.hs

```diff
@@ -273,12 +277,24 @@ liftCPErr = either internalError' return
 
 
 -- | Note: The BlockHeader param here is the PARENT HEADER of the new block-to-be
-execNewBlock :: MemPoolAccess -> BlockHeader -> MinerInfo -> PactServiceM PayloadWithOutputs
-execNewBlock memPoolAccess header miner = do
+execNewBlock
+    :: MemPoolAccess
+    -> BlockHeader
+    -> MinerInfo
+    -> PactServiceM PayloadWithOutputs
+execNewBlock mpAccess header miner = do
     let bHeight@(BlockHeight bh) = _blockHeight header
         bHash = _blockHash header
 
-    newTrans <- liftIO $! memPoolAccess bHeight bHash header
+    logInfo $ "execNewBlock, about to get call processFork: "
+           <> " (height = " <> sshow bHeight <> ")"
+           <> " (hash = " <> sshow bHash <> ")"
+    liftIO $ mpaProcessFork mpAccess header
```

Comment:
Why doesn't `getBlock` just call `processFork` internally? Not sure why you had to change the API here.

Reply:
Mostly because I was not certain if they would always be called together (i.e., there might be a better place for processFork to be called)